internal/backend/utils.go

```diff
@@ -21,6 +24,15 @@ func LoadAll(ctx context.Context, buf []byte, be restic.Backend, h restic.Handle
 			return cerr
 		}
 		buf = wr.Bytes()
+
+		if !retriedInvalidData && h.Type != restic.ConfigFile {
```

Comment:
The idea here is that at most once restic will check the returned data and make sure it matches the filename, if that's not the case an error will be returned so the download is retried. If that happens again the second time, the (invalid) data is returned so the caller can decide what to do with it. Correct?

Reply:
Essentially yes. It's rather unlikely that reading data a third time will make much of a difference of the first two tries returned damaged data. It is also necessary to e.g. let `debug examine` work, which downloads the full pack file before analyzing.

Reply:
I've added a comment.